config/routes.rb

```diff
@@ -1,4 +1,10 @@
 Rails.application.routes.draw do
+
+  resources :items do
+    resources :reservations, only: [:create]
+  end
+  get 'reservations', to: 'reservations#show'
+  post '/reservations', to: 'reservations#create'
```

Comment:
the API routes should follow best practices for example instead of looking into the reservation like this `http://127.0.0.1:3000/reservations` it should be like this `http://127.0.0.1:3000/api/v1/reservations`
you can check out the development branch `routes.rb` for reference